File: pdu/python/sample_action_msgs/pdu_cdr_conv_FibonacciActionFeedback.py
from .pdu_pytype_FibonacciActionFeedback import FibonacciActionFeedback
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_action_msgs.pdu_cdr_conv_ActionFeedbackHeader import *
from ..sample_action_msgs.pdu_cdr_conv_FibonacciFeedback import *



def py_to_cdr_body_FibonacciActionFeedback(writer: CdrWriter, src: FibonacciActionFeedback):
    py_to_cdr_body_ActionFeedbackHeader(writer, src.header)
    py_to_cdr_body_FibonacciFeedback(writer, src.body)


def cdr_body_to_py_FibonacciActionFeedback(reader: CdrReader, dst: FibonacciActionFeedback):
    cdr_body_to_py_ActionFeedbackHeader(reader, dst.header)
    cdr_body_to_py_FibonacciFeedback(reader, dst.body)
    return dst


def py_to_cdr_FibonacciActionFeedback(src: FibonacciActionFeedback) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_FibonacciActionFeedback(writer, src)
    return writer.bytes()


def cdr_to_py_FibonacciActionFeedback(cdr_payload) -> FibonacciActionFeedback:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = FibonacciActionFeedback()
    return cdr_body_to_py_FibonacciActionFeedback(reader, dst)

File: pdu/javascript/sample_action_msgs/pdu_cdr_conv_FibonacciActionFeedback.js
import { FibonacciActionFeedback } from './pdu_jstype_FibonacciActionFeedback.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { ActionFeedbackHeader } from '../hako_action_msgs/pdu_jstype_ActionFeedbackHeader.js';
import { PduActionFeedbackHeaderConverter } from '../hako_action_msgs/pdu_cdr_conv_ActionFeedbackHeader.js';
import { FibonacciFeedback } from '../sample_action_msgs/pdu_jstype_FibonacciFeedback.js';
import { PduFibonacciFeedbackConverter } from '../sample_action_msgs/pdu_cdr_conv_FibonacciFeedback.js';


export class PduFibonacciActionFeedbackConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { FibonacciActionFeedback } src
     */
    static to_cdr_body(writer, src) {
        PduActionFeedbackHeaderConverter.to_cdr_body(writer, src.header);
        PduFibonacciFeedbackConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { FibonacciActionFeedback } dst
     * @returns { FibonacciActionFeedback }
     */
    static cdr_body_to_js(reader, dst) {
        PduActionFeedbackHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduFibonacciFeedbackConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { FibonacciActionFeedback } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { FibonacciActionFeedback }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new FibonacciActionFeedback());
    }
}
